Model/AdyenInitPayments.php

```diff
@@ -144,21 +171,28 @@ public function execute(
         }
 
         $stateData = json_decode($stateData, true);
+
         // Validate JSON that has just been parsed if it was in a valid format
         if (json_last_error() !== JSON_ERROR_NONE) {
             throw new ValidatorException(
                 __('Payments call failed because stateData was not a valid JSON!')
             );
         }
+
+        // Validate payment method
+        if (!isset($stateData['paymentMethod']['type']) || $stateData['paymentMethod']['type'] !== 'paypal') {
+            throw new ValidatorException(
+                __('Invalid payment method. Only PayPal payments are allowed.')
```

Comment:
Let's not give context here and throw generic error message `Error with payment method please select different payment method.`.  We already consider usage of different payment methods malicious. Let's not forward them to the correct direction.